src/Messages/Interactions/TSMessage.m

```diff
@@ -102,6 +102,7 @@ - (instancetype)initWithTimestamp:(uint64_t)timestamp
     _expiresInSeconds = expiresInSeconds;
     _expireStartedAt = expireStartedAt;
     [self updateExpiresAt];
+    _receivedAtDate = [NSDate date];
```

Comment:
I think this is the centerpiece of the change, right?

By tracking local clock time of creation, instead of envelope timestamp, we can order the creation of these interactions.

(unless the user changes their clock)

Does this do the right thing for sync transcript messages (e.g. messages that you sent via desktop) received late? I guess if the server takes care to deliver us the messages in the order it received them?

One nit:  can we call it `createdAtDate` since the existing name is so close to the similar but different `receivedAt`?